src/expr/type_node.cpp

```diff
@@ -86,6 +87,16 @@ bool TypeNode::isInterpretedFinite() const {
   }
 }
 
+bool TypeNode::isFirstClass() const {
+  return ( getKind() != kind::FUNCTION_TYPE || options::ufHo() ) && 
```

Comment:
This seems a bit fragile - how do we ensure that this function stays correct?  Maybe something worth discussing or opening an issue for - not necessarily to be addressed by this PR.

Reply:
Sure, I suppose we could add this as an attribute in the kinds file.
Notice we have the same problem with maintaining "isSubtypeOf".

Reply:
Do you mind opening an issue for this?

Reply:
I've opened #1223.